screens/search/Search.tsx

```diff
@@ -0,0 +1,113 @@
+import {
+  Box,
+  Button,
+  Center,
+  Divider,
+  HStack,
+  ScrollView,
+  useColorModeValue,
+  VStack,
+} from 'native-base';
+import SearchBar from '../../components/searchbar/SearchBar';
+import { useEffect, useState } from 'react';
+import { DebounceSession } from '../../utils/utils';
+import { UserSearchResult } from '../../xplat/api';
+import { useSearchSubstringMatchers } from '../../utils/queries';
+import UserRow from '../../components/profile/UserRow';
+import ArchivedRouteRow from '../../components/route/ArchivedRouteRow';
+
+type SearchTab = 'routes' | 'users';
+
+const Search = () => {
+  const [query, setQuery] = useState<string>('');
+  const [userSearchResults, setUserSearchResults] = useState<
+    UserSearchResult[]
+  >([]);
+  const [archivedRoutesSearchResults, setArchivedRoutesSearchResults] =
+    useState<String[]>([]);
+  const [tabViewed, setTabViewed] = useState<SearchTab>('users');
+
+  const baseBgColor = useColorModeValue('lightMode.base', 'darkMode.base');
+
+  const { isLoading, isError, data, error } = useSearchSubstringMatchers();
+
+  useEffect(() => {
+    if (!data) return;
```

Comment:
Lets do explicit `undefined` check here. Data could theoretically be `false`.
```suggestion
    if (data === undefined) return;
```